niri-config/src/lib.rs

```diff
@@ -2370,6 +2370,65 @@ pub enum PreviewRender {
     ScreenCapture,
 }
 
+#[derive(Debug, Clone)]
+pub enum ConfigPath {
```

Comment:
Hm, perhaps this whole enum should live in `main.rs`? Seeing how it mainly deals with creating the right files and such. Then maybe some creation logic can also be inlined into its functions, simplifying them.

Especially using `maybe_create` to output an error feels kinda weird idk

Reply:
> Although `niri-config` is _not_ a public crate, and we don't encourage external use of it, others writing Rust-based tooling that is very tightly coupled to niri for their own systems might want to load the niri config; in which case, the precise logic of when which specific config path is loaded or when it "should" be created, can be quite useful indeed.

Well, this is kind of a theoretical concern for now; the practical concern though is that this would require adding various xdg directories deps to niri-config which it doesn't have right now, so it would be cleaner to leave all file path manipulation in main.rs. If this becomes a concern we can move things to niri-config as needed

Reply:
I don't think it requires adding new dependencies to `niri-config`. I only want it to include the logic for when to use/crate which path. The logic for *what paths should be considered* (e.g. loading xdg dirs) is still in `main.rs`, because this depends on the exact CLI interface of the code that is consuming `niri-config`.

Reply:
I moved the creation logic into `niri-config`. The interface on `ConfigPath` is much cleaner now: just `ConfigPath::load(&self)` and `ConfigPath::load_or_create(&self)`.

This also results in nicer errors.